external/flume/src/test/scala/org/apache/spark/streaming/flume/FlumeStreamSuite.scala

```diff
@@ -76,7 +75,7 @@ class FlumeStreamSuite extends FunSuite with BeforeAndAfter with Matchers with L
 
   /** Find a free port */
   private def findFreePort(): Int = {
-    Utils.startServiceOnPort(23456, (trialPort: Int) => {
+    Utils.startServiceOnPort((math.random * Int.MaxValue).toInt, (trialPort: Int) => {
```

Comment:
It will actually fail if the very large random port, plus offset - 1024, rolls over to a negative value. The returned port will be negative. Although it's a rare case, I think writing the correct version is also clearer.